### ISSUE TITLE
Parameters for INRIA dataset

### ISSUE THREAD

**Issue by raphaelsulzer** [HeinzJiao/PolyR-CNN#6]
Hi and thank you for making this nice work available here.

I am trying to train your method on the INRIA dataset. Your paper mentions that you crop the 5000x5000 images to 512x512 patches, similar to HiSup. In `preprocess_annotations.json` is seems like that the image size is set to 300 px. Do I just change that to 512 px to preprocess the INRIA dataset? Further, the swin transformer backbone takes images of size 224x224 if I am not mistaken. Do you just resize / downsample the cropped INRIA images to this size? 

Besides the image size, your paper mentions that you use unified vertex number per polygon `M=50` for the INRIA dataset. If I set `M=50` during preprocessing, I later get an error  somewhere in your forward pass. What should I use?

**Comment by raphaelsulzer** [comment]
I trained a model with the config below. Even after 100 epochs the loss_polygon is around 10, loss_gio around 0.3 and loss_cor around 2. 

Below some predictions on the validation set produced with the demo.py (I slightly changed the visualisation).

Any idea what could have gone wrong? Or are those the expected results?

![Image](https://github.com/user-attachments/assets/51d46e8a-7c90-4e7c-ad63-a1af0904f0a1)
![Image](https://github.com/user-attachments/assets/62140032-ec81-4e6a-b288-b65698ec8bf5)

```
_BASE_: "Base-PolyRCNN.yaml"
DATALOADER:
  NUM_WORKERS: 1
MODEL:
  WEIGHTS: "detectron2://ImageNetPretrained/torchvision/R-50.pkl"
  RESNETS:
    DEPTH: 50
    STRIDE_IN_1X1: False
  PolyRCNN:
    NUM_PROPOSALS: 100
    NUM_CLASSES: 1
DATASETS:
  TRAIN: ("inria_train",)  # 89280 images in total (outer polys), 10908 images in total (inner polys)
  TEST:  ("inria_val",)
SOLVER:
  STEPS: (60698, 68200)  # 89th epoch
  MAX_ITER: 68201  # 100th epoch
  CHECKPOINT_PERIOD: 682  # save checkpoint per epoch
  IMS_PER_BATCH: 16
TEST:
  EVAL_PERIOD: 1000
INPUT:
  CROP:
    ENABLED: False
  FORMAT: "RGB"
```

**Comment by HeinzJiao** [comment]
Thank you very much for your interest in our work and for sharing your results.

Training on the Inria dataset is more challenging than CrowdAI, as it contains complex building structures with internal holes, which are typically better handled by segmentation-based methods. PolyR-CNN was originally designed for datasets like CrowdAI, where each building is represented by a single polygon, and the structures are generally simpler.

Regarding training, while the number of epochs provided in the paper serves as a general reference, the optimal convergence time may vary depending on the computational setup. In our experiments, if I recall correctly, loss_polygon eventually reached around 6. You may consider extending the training duration (particularly before learning rate decay) and monitoring whether the loss continues to decrease. The final results should resemble those presented in Figure 6, though we also observed that performance may degrade in certain challenging cases. Post-processing techniques, such as merging nearly parallel edges adopted in HiSup, can help refine jagged boundaries.

Another key consideration is that the current released version of PolyR-CNN predicts only a single polygon per building, whereas the Inria dataset contains both external building contours and internal holes. In our experiments, we modified PolyR-CNN to distinguish between these as separate classes, but this version has not yet been finalized for release. If using the current code, a possible workaround is to first remove internal holes and train the model on external building outlines only, then train a separate model for internal holes and merge the results in post-processing. The provided preprocess_annotations.json script is designed for datasets without internal holes but can be adapted to process external outlines and internal contours separately.

Hope this helps!

**Comment by HeinzJiao** [comment]
> Hi and thank you for making this nice work available here.
> 
> I am trying to train your method on the INRIA dataset. Your paper mentions that you crop the 5000x5000 images to 512x512 patches, similar to HiSup. In `preprocess_annotations.json` is seems like that the image size is set to 300 px. Do I just change that to 512 px to preprocess the INRIA dataset? Further, the swin transformer backbone takes images of size 224x224 if I am not mistaken. Do you just resize / downsample the cropped INRIA images to this size?
> 
> Besides the image size, your paper mentions that you use unified vertex number per polygon `M=50` for the INRIA dataset. If I set `M=50` during preprocessing, I later get an error somewhere in your forward pass. What should I use?

To use preprocess_annotations.json with a different image size, you can simply change it to 512 px to match the INRIA dataset. Regarding the image size adaptation for Swin Transformer, I did not specifically modify it, as I directly adopted the backbone implementation from other works. You may check ./polyrcnn/swin_transformer.py for more details on how the image resizing is handled.

After closely reviewing your visualized results, I noticed that you might be using a larger M value than M = 50, which was specified in our setup. To use M = 50, NUM_CORNERS in ./polyrcnn/config.py (inside the function def add_polyrcnn_config(cfg):) needs to be manually adjusted, as it is not automatically set. I apologize for not specifying this more clearly on the GitHub page.

Using a higher M value on Inria, given the dataset’s structural complexity, could increase the difficulty of vertex classification, potentially affecting performance and leading to longer convergence times. It might be helpful to check this parameter and see if adjusting it improves your results.

**Comment by hcscysh** [comment]
@raphaelsulzer do you meet the same problem ? when I run the demo.py ,the config file is polyrcnn.res50.100pro.aicrowd.yaml.I get a bug as following:
Some model parameters or buffers are not found in the checkpoint:
backbone.fpn_lateral2.{bias, weight}
backbone.fpn_lateral3.{bias, weight}
backbone.fpn_lateral4.{bias, weight}
backbone.fpn_lateral5.{bias, weight}
backbone.fpn_output2.{bias, weight}
backbone.fpn_output3.{bias, weight}
backbone.fpn_output4.{bias, weight}
backbone.fpn_output5.{bias, weight}
criterion.empty_weight
head.head_series.0.bboxes_delta.{bias, weight}
head.head_series.0.class_logits.{bias, weight}
head.head_series.0.cls_module.0.weight
head.head_series.0.cls_module.1.{bias, weight}
head.head_series.0.cor_module.0.weight
head.head_series.0.cor_module.1.{bias, weight}
head.head_series.0.cor_module.3.weight
head.head_series.0.cor_module.4.{bias, weight}
head.head_series.0.cor_module.6.weight
head.head_series.0.cor_module.7.{bias, weight}
head.head_series.0.corner_logits.{bias, weight}
head.head_series.0.corner_mlp.0.{bias, weight}
head.head_series.0.corner_mlp.2.{bias, weight}
head.head_series.0.inst_interact.dynamic_layer.{bias, weight}
head.head_series.0.inst_interact.norm1.{bias, weight}
head.head_series.0.inst_interact.norm2.{bias, weight}
head.head_series.0.inst_interact.norm3.{bias, weight}
head.head_series.0.inst_interact.out_layer.{bias, weight}
head.head_series.0.linear1.{bias, weight}
head.head_series.0.linear2.{bias, weight}
head.head_series.0.norm1.{bias, weight}
head.head_series.0.norm2.{bias, weight}
head.head_series.0.norm3.{bias, weight}
head.head_series.0.norm4.{bias, weight}
head.head_series.0.pol_module.0.weight
head.head_series.0.pol_module.1.{bias, weight}
head.head_series.0.pol_module.3.weight
head.head_series.0.pol_module.4.{bias, weight}
head.head_series.0.pol_module.6.weight
head.head_series.0.pol_module.7.{bias, weight}
head.head_series.0.polygons_delta.{bias, weight}
head.head_series.0.reg_module.0.weight
head.head_series.0.reg_module.1.{bias, weight}
head.head_series.0.reg_module.3.weight
head.head_series.0.reg_module.4.{bias, weight}
head.head_series.0.reg_module.6.weight
head.head_series.0.reg_module.7.{bias, weight}
head.head_series.0.self_attn.out_proj.{bias, weight}
head.head_series.0.self_attn.{in_proj_bias, in_proj_weight}
head.head_series.0.self_attn1.out_proj.{bias, weight}
head.head_series.0.self_attn1.{in_proj_bias, in_proj_weight}
head.head_series.1.bboxes_delta.{bias, weight}
head.head_series.1.class_logits.{bias, weight}
head.head_series.1.cls_module.0.weight
head.head_series.1.cls_module.1.{bias, weight}
head.head_series.1.cor_module.0.weight
head.head_series.1.cor_module.1.{bias, weight}
head.head_series.1.cor_module.3.weight
head.head_series.1.cor_module.4.{bias, weight}
head.head_series.1.cor_module.6.weight
head.head_series.1.cor_module.7.{bias, weight}
head.head_series.1.corner_logits.{bias, weight}
head.head_series.1.corner_mlp.0.{bias, weight}
head.head_series.1.corner_mlp.2.{bias, weight}
head.head_series.1.inst_interact.dynamic_layer.{bias, weight}
head.head_series.1.inst_interact.norm1.{bias, weight}
head.head_series.1.inst_interact.norm2.{bias, weight}
head.head_series.1.inst_interact.norm3.{bias, weight}
head.head_series.1.inst_interact.out_layer.{bias, weight}
head.head_series.1.linear1.{bias, weight}
head.head_series.1.linear2.{bias, weight}
head.head_series.1.norm1.{bias, weight}
head.head_series.1.norm2.{bias, weight}
head.head_series.1.norm3.{bias, weight}
head.head_series.1.norm4.{bias, weight}
head.head_series.1.pol_module.0.weight
head.head_series.1.pol_module.1.{bias, weight}
head.head_series.1.pol_module.3.weight
head.head_series.1.pol_module.4.{bias, weight}
head.head_series.1.pol_module.6.weight
head.head_series.1.pol_module.7.{bias, weight}
head.head_series.1.polygons_delta.{bias, weight}
head.head_series.1.reg_module.0.weight
head.head_series.1.reg_module.1.{bias, weight}
head.head_series.1.reg_module.3.weight
head.head_series.1.reg_module.4.{bias, weight}
head.head_series.1.reg_module.6.weight
head.head_series.1.reg_module.7.{bias, weight}
head.head_series.1.self_attn.out_proj.{bias, weight}
head.head_series.1.self_attn.{in_proj_bias, in_proj_weight}
head.head_series.1.self_attn1.out_proj.{bias, weight}
head.head_series.1.self_attn1.{in_proj_bias, in_proj_weight}
head.head_series.2.bboxes_delta.{bias, weight}
head.head_series.2.class_logits.{bias, weight}
head.head_series.2.cls_module.0.weight
head.head_series.2.cls_module.1.{bias, weight}
head.head_series.2.cor_module.0.weight
head.head_series.2.cor_module.1.{bias, weight}
head.head_series.2.cor_module.3.weight
head.head_series.2.cor_module.4.{bias, weight}
head.head_series.2.cor_module.6.weight
head.head_series.2.cor_module.7.{bias, weight}
head.head_series.2.corner_logits.{bias, weight}
head.head_series.2.corner_mlp.0.{bias, weight}
head.head_series.2.corner_mlp.2.{bias, weight}
head.head_series.2.inst_interact.dynamic_layer.{bias, weight}
head.head_series.2.inst_interact.norm1.{bias, weight}
head.head_series.2.inst_interact.norm2.{bias, weight}
head.head_series.2.inst_interact.norm3.{bias, weight}
head.head_series.2.inst_interact.out_layer.{bias, weight}
head.head_series.2.linear1.{bias, weight}
head.head_series.2.linear2.{bias, weight}
head.head_series.2.norm1.{bias, weight}
head.head_series.2.norm2.{bias, weight}
head.head_series.2.norm3.{bias, weight}
head.head_series.2.norm4.{bias, weight}
head.head_series.2.pol_module.0.weight
head.head_series.2.pol_module.1.{bias, weight}
head.head_series.2.pol_module.3.weight
head.head_series.2.pol_module.4.{bias, weight}
head.head_series.2.pol_module.6.weight
head.head_series.2.pol_module.7.{bias, weight}
head.head_series.2.polygons_delta.{bias, weight}
head.head_series.2.reg_module.0.weight
head.head_series.2.reg_module.1.{bias, weight}
head.head_series.2.reg_module.3.weight
head.head_series.2.reg_module.4.{bias, weight}
head.head_series.2.reg_module.6.weight
head.head_series.2.reg_module.7.{bias, weight}
head.head_series.2.self_attn.out_proj.{bias, weight}
head.head_series.2.self_attn.{in_proj_bias, in_proj_weight}
head.head_series.2.self_attn1.out_proj.{bias, weight}
head.head_series.2.self_attn1.{in_proj_bias, in_proj_weight}
head.head_series.3.bboxes_delta.{bias, weight}
head.head_series.3.class_logits.{bias, weight}
head.head_series.3.cls_module.0.weight
head.head_series.3.cls_module.1.{bias, weight}
head.head_series.3.cor_module.0.weight
head.head_series.3.cor_module.1.{bias, weight}
head.head_series.3.cor_module.3.weight
head.head_series.3.cor_module.4.{bias, weight}
head.head_series.3.cor_module.6.weight
head.head_series.3.cor_module.7.{bias, weight}
head.head_series.3.corner_logits.{bias, weight}
head.head_series.3.corner_mlp.0.{bias, weight}
head.head_series.3.corner_mlp.2.{bias, weight}
head.head_series.3.inst_interact.dynamic_layer.{bias, weight}
head.head_series.3.inst_interact.norm1.{bias, weight}
head.head_series.3.inst_interact.norm2.{bias, weight}
head.head_series.3.inst_interact.norm3.{bias, weight}
head.head_series.3.inst_interact.out_layer.{bias, weight}
head.head_series.3.linear1.{bias, weight}
head.head_series.3.linear2.{bias, weight}
head.head_series.3.norm1.{bias, weight}
head.head_series.3.norm2.{bias, weight}
head.head_series.3.norm3.{bias, weight}
head.head_series.3.norm4.{bias, weight}
head.head_series.3.pol_module.0.weight
head.head_series.3.pol_module.1.{bias, weight}
head.head_series.3.pol_module.3.weight
head.head_series.3.pol_module.4.{bias, weight}
head.head_series.3.pol_module.6.weight
head.head_series.3.pol_module.7.{bias, weight}
head.head_series.3.polygons_delta.{bias, weight}
head.head_series.3.reg_module.0.weight
head.head_series.3.reg_module.1.{bias, weight}
head.head_series.3.reg_module.3.weight
head.head_series.3.reg_module.4.{bias, weight}
head.head_series.3.reg_module.6.weight
head.head_series.3.reg_module.7.{bias, weight}
head.head_series.3.self_attn.out_proj.{bias, weight}
head.head_series.3.self_attn.{in_proj_bias, in_proj_weight}
head.head_series.3.self_attn1.out_proj.{bias, weight}
head.head_series.3.self_attn1.{in_proj_bias, in_proj_weight}
head.head_series.4.bboxes_delta.{bias, weight}
head.head_series.4.class_logits.{bias, weight}
head.head_series.4.cls_module.0.weight
head.head_series.4.cls_module.1.{bias, weight}
head.head_series.4.cor_module.0.weight
head.head_series.4.cor_module.1.{bias, weight}
head.head_series.4.cor_module.3.weight
head.head_series.4.cor_module.4.{bias, weight}
head.head_series.4.cor_module.6.weight
head.head_series.4.cor_module.7.{bias, weight}
head.head_series.4.corner_logits.{bias, weight}
head.head_series.4.corner_mlp.0.{bias, weight}
head.head_series.4.corner_mlp.2.{bias, weight}
head.head_series.4.inst_interact.dynamic_layer.{bias, weight}
head.head_series.4.inst_interact.norm1.{bias, weight}
head.head_series.4.inst_interact.norm2.{bias, weight}
head.head_series.4.inst_interact.norm3.{bias, weight}
head.head_series.4.inst_interact.out_layer.{bias, weight}
head.head_series.4.linear1.{bias, weight}
head.head_series.4.linear2.{bias, weight}
head.head_series.4.norm1.{bias, weight}
head.head_series.4.norm2.{bias, weight}
head.head_series.4.norm3.{bias, weight}
head.head_series.4.norm4.{bias, weight}
head.head_series.4.pol_module.0.weight
head.head_series.4.pol_module.1.{bias, weight}
head.head_series.4.pol_module.3.weight
head.head_series.4.pol_module.4.{bias, weight}
head.head_series.4.pol_module.6.weight
head.head_series.4.pol_module.7.{bias, weight}
head.head_series.4.polygons_delta.{bias, weight}
head.head_series.4.reg_module.0.weight
head.head_series.4.reg_module.1.{bias, weight}
head.head_series.4.reg_module.3.weight
head.head_series.4.reg_module.4.{bias, weight}
head.head_series.4.reg_module.6.weight
head.head_series.4.reg_module.7.{bias, weight}
head.head_series.4.self_attn.out_proj.{bias, weight}
head.head_series.4.self_attn.{in_proj_bias, in_proj_weight}
head.head_series.4.self_attn1.out_proj.{bias, weight}
head.head_series.4.self_attn1.{in_proj_bias, in_proj_weight}
head.head_series.5.bboxes_delta.{bias, weight}
head.head_series.5.class_logits.{bias, weight}
head.head_series.5.cls_module.0.weight
head.head_series.5.cls_module.1.{bias, weight}
head.head_series.5.cor_module.0.weight
head.head_series.5.cor_module.1.{bias, weight}
head.head_series.5.cor_module.3.weight
head.head_series.5.cor_module.4.{bias, weight}
head.head_series.5.cor_module.6.weight
head.head_series.5.cor_module.7.{bias, weight}
head.head_series.5.corner_logits.{bias, weight}
head.head_series.5.corner_mlp.0.{bias, weight}
head.head_series.5.corner_mlp.2.{bias, weight}
head.head_series.5.inst_interact.dynamic_layer.{bias, weight}
head.head_series.5.inst_interact.norm1.{bias, weight}
head.head_series.5.inst_interact.norm2.{bias, weight}
head.head_series.5.inst_interact.norm3.{bias, weight}
head.head_series.5.inst_interact.out_layer.{bias, weight}
head.head_series.5.linear1.{bias, weight}
head.head_series.5.linear2.{bias, weight}
head.head_series.5.norm1.{bias, weight}
head.head_series.5.norm2.{bias, weight}
head.head_series.5.norm3.{bias, weight}
head.head_series.5.norm4.{bias, weight}
head.head_series.5.pol_module.0.weight
head.head_series.5.pol_module.1.{bias, weight}
head.head_series.5.pol_module.3.weight
head.head_series.5.pol_module.4.{bias, weight}
head.head_series.5.pol_module.6.weight
head.head_series.5.pol_module.7.{bias, weight}
head.head_series.5.polygons_delta.{bias, weight}
head.head_series.5.reg_module.0.weight
head.head_series.5.reg_module.1.{bias, weight}
head.head_series.5.reg_module.3.weight
head.head_series.5.reg_module.4.{bias, weight}
head.head_series.5.reg_module.6.weight
head.head_series.5.reg_module.7.{bias, weight}
head.head_series.5.self_attn.out_proj.{bias, weight}
head.head_series.5.self_attn.{in_proj_bias, in_proj_weight}
head.head_series.5.self_attn1.out_proj.{bias, weight}
head.head_series.5.self_attn1.{in_proj_bias, in_proj_weight}
init_proposal_boxes.weight
init_proposal_polygons.weight
The checkpoint state_dict contains keys that are not used by the model:
  stem.fc.{bias, weight}
